generator/src/main/java/com/scottlogic/deg/generator/generation/string/RegexStringGenerator.java

```diff
@@ -408,28 +421,42 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(this.automaton, this.getClass());
     }
```

Comment:
I believe these should be less than or equal to in both cases, rather than less than (consider substituting `a`, `b` and `Long.MAX_VALUE` with small values to verify).

Probably unnecessary here but would make any later implementations trivial and less brittle to change between one of these functions - we could replace the single different operator with a function passed in, giving us:

`multiplyWouldOverflow(long a, long b) { return checkOverflow((max, b) -> max / b) }`
`addWouldOverflow(long a, long b) { return checkOverflow((max, b) -> max - b) }`